Mathlib/Topology/MetricSpace/Algebra.lean

```diff
@@ -200,3 +200,40 @@ instance [AddMonoid α] [LipschitzAdd α] : LipschitzMul (Multiplicative α) :=
 @[to_additive]
 instance [Monoid α] [LipschitzMul α] : LipschitzMul αᵒᵈ :=
   ‹LipschitzMul α›
+
+variable {ι : Type*} [Fintype ι]
+
+instance Pi.instBoundedSMul {α : Type*} {β : ι → Type*} [PseudoMetricSpace α]
+    [∀ i, PseudoMetricSpace (β i)] [Zero α] [∀ i, Zero (β i)] [∀ i, SMul α (β i)]
+    [∀ i, BoundedSMul α (β i)] : BoundedSMul α (∀ i, β i) where
+  dist_smul_pair' x y₁ y₂ :=
+    (dist_pi_le_iff <| by positivity).2 fun i ↦
+      (dist_smul_pair _ _ _).trans <| mul_le_mul_of_nonneg_left (dist_le_pi_dist _ _ _) dist_nonneg
+  dist_pair_smul' x₁ x₂ y :=
+    (dist_pi_le_iff <| by positivity).2 fun i ↦
+      (dist_pair_smul _ _ _).trans <| mul_le_mul_of_nonneg_left (dist_le_pi_dist _ 0 _) dist_nonneg
+
+instance Pi.instBoundedSMul' {α β : ι → Type*} [∀ i, PseudoMetricSpace (α i)]
+    [∀ i, PseudoMetricSpace (β i)] [∀ i, Zero (α i)] [∀ i, Zero (β i)] [∀ i, SMul (α i) (β i)]
+    [∀ i, BoundedSMul (α i) (β i)] : BoundedSMul (∀ i, α i) (∀ i, β i) where
+  dist_smul_pair' x y₁ y₂ :=
+    (dist_pi_le_iff <| by positivity).2 fun i ↦
+      (dist_smul_pair _ _ _).trans <|
+        mul_le_mul (dist_le_pi_dist _ 0 _) (dist_le_pi_dist _ _ _) dist_nonneg dist_nonneg
+  dist_pair_smul' x₁ x₂ y :=
+    (dist_pi_le_iff <| by positivity).2 fun i ↦
+      (dist_pair_smul _ _ _).trans <|
+        mul_le_mul (dist_le_pi_dist _ _ _) (dist_le_pi_dist _ 0 _) dist_nonneg dist_nonneg
```

Comment:
This would be a lot easier if we went via an `nndist` constructor, as all the `nonneg` constraints would vanish.
That's obviously out of scope here, but if you feel like being sniped by a refactor...

bors d+

Reply:
I will look into it